pages/plans/[id].js

```diff
@@ -22,60 +32,42 @@ const PlanSummary = ({ hackneyToken, plan }) => {
     return baseString;
   };
 
-  const updatePlan = async newPlan => {
-    setPlan(newPlan);
-    setEditGoal(false);
-  };
-
   return (
     <>
       <h1>{getPossessiveName(firstName, lastName)} shared plan</h1>
       {editGoal && (
         <AddGoal
-          hackneyToken={hackneyToken}
-          plan={_plan}
-          updatePlan={updatePlan}
+          goal={goal}
+          onGoalAdded={async goal => {
+            await addGoal(goal);
+            setEditGoal(false);
+          }}
         />
       )}
-      {!editGoal && <GoalSummary hackneyToken={hackneyToken} plan={_plan} />}
+      {!editGoal && <GoalSummary plan={plan} />}
       {!editGoal && (
         <Button text="Edit goal" onClick={() => setEditGoal(true)} />
       )}
-      {!editGoal && (
-        <AddAction
-          hackneyToken={hackneyToken}
-          id={id}
-          updatePlan={updatePlan}
-        />
-      )}
+      <ActionsList actions={plan.goal?.actions || []} />
+      {!editGoal && <AddAction id={id} onActionAdded={addAction} />}
       {!editGoal && goal && goal.useAsPhp && <LegalText />}
     </>
   );
 };
 
-PlanSummary.getInitialProps = async ({ query, req, res }) => {
-  const hackneyToken = getToken(req);
+PlanSummary.getInitialProps = async ({ query: { id }, req, res }) => {
+  try {
+    const token = getToken(req);
+    const plan = await requestPlan(id, { token });
 
-  const response = await fetch(
-    `${process.env.SHARED_PLAN_API_URL}/plans/${query.id}`,
-    {
-      headers: {
-        'Content-Type': 'application/json',
-        Authorization: `Bearer ${hackneyToken}`
-      }
-    }
-  );
-  if (response.status === 404) {
-    res.statusCode = 404;
-    return res.end('Not found');
+    return {
+      planId: id,
```

Comment:
Nitpicking but is planId required here or can we just extract it from initalPlan

Reply:
Yeah I think we probably could do that, was trying to account for scenarios where for some reason the plan didn't load server-side and we could retry it client-side. But probably over-engineering a bit given the likelihood of that.

Reply:
Will merge for now and look to address in a future PR.